sdv/data_processing/data_processor.py

```diff
@@ -202,20 +205,62 @@ def create_anonymized_transformer(self, sdtype, column_metadata):
 
         return get_anonymized_transformer(sdtype, kwargs)
 
+    def create_primary_key_transformer(self, sdtype, column_metadata):
+        """Create an instance for the primary key.
+
+        Read the keyword arguments from the ``column_metadata`` and use them to create
+        an instance of an ``RegexGenerator`` or ``AnonymizedFaker`` transformer with
+        ``enforce_uniqueness`` set to ``True``.
+
+        Args:
+            sdtype (str):
+                Sematic data type or a ``Faker`` function name.
+            column_metadata (dict):
+                A dictionary representing the rest of the metadata for the given ``sdtype``.
+
+        Returns:
+            transformer:
+                Instance of ``rdt.transformers.text.RegexGenerator`` or
+                ``rdt.transformers.pii.AnonymizedFaker`` with ``enforce_uniqueness`` set to
+                ``True``.
+        """
+        regex_format = column_metadata.get('regex_format')
+        if regex_format:
+            transformer = rdt.transformers.RegexGenerator(
+                regex_format=regex_format,
+                enforce_uniqueness=True
+            )
+
+        else:
+            column_metadata['enforce_uniqueness'] = True
+            transformer = self.create_anonymized_transformer(sdtype, column_metadata)
+
+        return transformer
+
     def _create_config(self, data, columns_created_by_constraints):
         sdtypes = {}
         transformers = {}
         self._anonymized_columns = []
+
+        self._primary_keys = []
+        if self.metadata._primary_key:
+            self._primary_keys = list(cast_to_iterable(self.metadata._primary_key))
+
         for column in set(data.columns) - columns_created_by_constraints:
             column_metadata = self.metadata._columns.get(column)
             sdtype = column_metadata.get('sdtype')
+            sdtypes[column] = sdtype
             if column_metadata.get('pii'):
```

Comment:
this is the same check as on line 259. Why not just move the line underneath to there?